pkg/collector/process.go

```diff
@@ -231,36 +242,27 @@ func (cp *collectingProcess) deleteTemplate(obsDomainID uint32, templateID uint1
 	delete(cp.templatesMap[obsDomainID], templateID)
 }
 
-// a common binary decoder to specified interfaces
-func decode(buffer io.Reader, outputs ...interface{}) error {
-	var err error
-	for _, out := range outputs {
-		err = binary.Read(buffer, binary.BigEndian, out)
-	}
-	return err
-}
-
-// get buffer length by decoding the header
+// getMessageLength returns buffer length by decoding the header
 func getMessageLength(msgBuffer *bytes.Buffer) (int, error) {
 	packet := entities.Message{}
 	var id, length uint16
-	err := decode(msgBuffer, &packet.Version, &packet.BufferLength, &packet.ExportTime, &packet.SeqNumber, &packet.ObsDomainID, &id, &length)
+	err := util.Decode(msgBuffer, &packet.Version, &packet.BufferLength, &packet.ExportTime, &packet.SeqNumber, &packet.ObsDomainID, &id, &length)
 	if err != nil {
 		return 0, fmt.Errorf("Cannot decode message: %v", err)
 	}
 	return int(packet.BufferLength), nil
 }
 
-// get string field length for data record
+// getFieldLength returns string field length for data record
 func getFieldLength(dataBuffer *bytes.Buffer) int {
 	lengthBuff := dataBuffer.Next(1)
 	var length1 uint8
```

Comment:
Referencing the link here with comments will be good.
https://tools.ietf.org/html/rfc7011#appendix-A.5